src/Common/OvercommitTracker.cpp

```diff
@@ -34,6 +54,8 @@ OvercommitResult OvercommitTracker::needToStopQuery(MemoryTracker * tracker, Int
     std::unique_lock<std::mutex> global_lock(global_mutex);
     std::unique_lock<std::mutex> lk(overcommit_m);
 
+    size_t id = next_id++;
```

Comment:
Should we make `next_id` atomic and use `fetch_add`?

Reply:
This code is always performed in a single thread, so there is no need to use atomic

Reply:
This code is called from
https://github.com/ClickHouse/ClickHouse/blob/60b9d81773d5a56486c5685ec3880d6caf207d1f/src/Common/MemoryTracker.cpp#L220-L221
And it does not look like single-threaded code at all

Reply:
Every public member function of OvercommitTracker starts with lock acquisition. So all code inside of OvercommitTracker is single-threaded. In `OvercommitTracker::needToStopQuery` lock acquired here:
https://github.com/ClickHouse/ClickHouse/blob/099055c1833224bf9212897a7a283a7cf529c803/src/Common/OvercommitTracker.cpp#L55